src/main/java/com/hrznstudio/sandbox/mixin/fabric/enchant/MixinEnchantmentHelper.java

```diff
@@ -17,35 +19,10 @@
 public abstract class MixinEnchantmentHelper {
 
     /**
-     * @author Coded
-     * @reason mojang was dumb
+     * @author B0undarybreaker
      */
-    @Overwrite
-    public static List<InfoEnchantment> getHighestApplicableEnchantmentsAtPower(int int_1, ItemStack itemStack_1, boolean treasureAllowed) {
-        List<InfoEnchantment> list_1 = Lists.newLinkedList();
-        boolean boolean_2 = itemStack_1.getItem() == Items.BOOK;
-        Iterator var6 = Registry.ENCHANTMENT.iterator();
-
-        while (true) {
-            while (true) {
-                Enchantment enchantment_1;
-                do {
-                    do {
-                        if (!var6.hasNext()) {
-                            return list_1;
-                        }
-
-                        enchantment_1 = (Enchantment) var6.next();
-                    } while (enchantment_1.isTreasure() && !treasureAllowed);
-                } while (!enchantment_1.isAcceptableItem(itemStack_1) && !boolean_2);
-
-                for (int int_2 = enchantment_1.getMaximumLevel(); int_2 > enchantment_1.getMinimumLevel() - 1; --int_2) {
-                    if (int_1 >= enchantment_1.getMinimumPower(int_2) && int_1 <= enchantment_1.method_20742(int_2)) {
-                        list_1.add(new InfoEnchantment(enchantment_1, int_2));
-                        break;
-                    }
-                }
-            }
-        }
+    @ModifyVariable(method = "getHighestApplicableEnchantmentsAtPower", at = @At(value = "INVOKE_ASSIGN", target = "Lcom/google/common/collect/Lists;newArrayList()Ljava/util/ArrayList;"))
+    private static List<InfoEnchantment> getEnchantmentList(List<InfoEnchantment> original) {
```

Comment:
So this injection doesnt seem to do the same as the overwrite, it was added to make it so it'd check the enchantment itself rather than the type if the input stack was valid
```java
} while(!enchantment_1.type.isAcceptableItem(item_1) && !boolean_2);```
to
```java
} while (!enchantment_1.isAcceptableItem(itemStack_1) && !boolean_2);```

Reply:
Ahhh, I misinterpreted what it was about. I can swap this to a `@Redirect` instead, so it would only conflict if someone else tried to change the call for this exact method, instead of trying to change stuff with the method.

Reply:
👍 sounds good